rundeckapp/grails-spa/packages/ui-trellis/src/library/components/plugins/pluginConfig.vue

```diff
@@ -135,353 +135,377 @@ import PluginInfo from './PluginInfo.vue'
 import PluginValidation from '../../interfaces/PluginValidation'
 import PluginPropView from './pluginPropView.vue'
 import PluginPropEdit from './pluginPropEdit.vue'
-import {cleanConfigInput,convertArrayInput} from '../../modules/InputUtils'
+import {Component, Prop, Watch} from 'vue-property-decorator'
+import {cleanConfigInput, convertArrayInput} from '../../modules/InputUtils'
 
 import {diff} from 'deep-object-diff'
 
-import {getPluginProvidersForService,
-  getServiceProviderDescription,
-  validatePluginConfig} from '../../modules/pluginService'
+import {
+    getPluginProvidersForService,
+    getServiceProviderDescription,
+    validatePluginConfig
+} from '../../modules/pluginService'
 
 interface PropGroup{
-  name?:string
-  secondary:boolean
-  props:any[],
+    name?:string
+    secondary:boolean
+    props:any[],
 }
 
-export default Vue.extend({
-  name: 'PluginConfig',
-  components: {
-    Expandable,
-    AceEditor,
-    PluginInfo,
-    PluginPropView,
-    PluginPropEdit
-  },
-  props: [
-    'serviceName',
-    'provider',
-    'config',
-    'mode',
-    'showTitle',
-    'showIcon',
-    'showDescription',
-    'value',
-    'savedProps',
-    'pluginConfig',
-    'validation',
-    'readOnly',
-    'validationWarningText',
-    'scope',
-    'defaultScope',
-    'contextAutocomplete',
-    'autocompleteCallback'
-  ],
-  data () {
-    return {
-      props: [] as any[],
-      detail: {},
-      error: null as any|null,
-      isReadOnly: this.readOnly!== null ? this.readOnly : false,
-      propsComputedSelectorData: {},
-      inputShowTitle: this.showTitle !== null ? this.showTitle : true,
-      inputShowIcon: this.showIcon !== null ? this.showIcon : true,
-      inputShowDescription: this.showDescription !== null ? this.showDescription : true,
-      inputValues: {} as any,
-      inputSaved: {} as any,
-      inputSavedProps: (typeof this.savedProps !== 'undefined') ? this.savedProps : ['type'],
-      rkey: 'r_' + Math.floor(Math.random() * Math.floor(1024)).toString(16) + '_',
-      groupExpand:{} as {[name:string]:boolean},
-      inputLoaded: false,
-      inputContextAutocomplete: this.contextAutocomplete !== null ? this.contextAutocomplete : false
+@Component({
```

Comment:
this needs to be reverted back to the "options API" style Vue component for now, we need this style to make it easier for the vue3 migration